### PR TITLE
Fix/updated state when buy item

### PR DESCRIPTION
**Pull request by AlefrankM**

### CODE REVIEW

**Review by juliswer — DISMISSED** [review]
Looks good

File: src/components/item-details/bid-tab/listing-tab-content.tsx [hunk 1]
```diff
@@ -33,27 +28,28 @@ export const ListingTabContent = ({ item }: Props) => {
   });
 
   const onSubmit = async ({ price }: TForm) => {
-    if (!item.listId) {
+    if (!selectedItem.listId) {
       dispatch(
         listItem({
```
Comment (mmiranda-stackit) [inline src/components/item-details/bid-tab/listing-tab-content.tsx:33]: Please add withProtection action here

File: src/components/item-details/bid-tab/listing-tab-content.tsx [hunk 2]
```diff
@@ -33,27 +28,28 @@ export const ListingTabContent = ({ item }: Props) => {
...
-          listId: item.listId,
+          tokenId: selectedItem.tokenId,
+          itemId: selectedItem.itemId,
+          ownerAddress: selectedItem.owner.address,
+          listId: selectedItem.listId,
         })
       );
     } else {
       dispatch(
         changePriceItem({
```
Comment (mmiranda-stackit) [inline src/components/item-details/bid-tab/listing-tab-content.tsx:44]: Please add withProtection action here

File: src/features/marketplace/store/marketplace.slice.ts [hunk 3]
```diff
@@ -69,6 +73,16 @@ const marketplaceSlice = createSlice({
       state.isLoading = false;
       state.isListed = false;
     });
+    builder.addCase(buyItem.pending, (state) => {
```
Comment (mmiranda-stackit) [inline src/features/marketplace/store/marketplace.slice.ts:76]: Unit tests ?

File: src/features/marketplace/store/marketplace.slice.ts [hunk 4]
```diff
@@ -55,6 +56,9 @@ const marketplaceSlice = createSlice({
     resetMarketplaceFilters: (state) => {
       state.marketplaceFilters = initialState.marketplaceFilters;
     },
+    setSelectedItem: (state, { payload }) => {
```
Comment (mmiranda-stackit) [inline src/features/marketplace/store/marketplace.slice.ts:59]: Unit tests?

File: src/features/leda-nft/store/leda-nft.actions.ts [hunk 5]
```diff
@@ -42,8 +42,10 @@ const mintNft = createAsyncThunk<Item | undefined, ItemRequest, { rejectValue: v
 
 const findAll = createAsyncThunk('nft/findAll', async () => itemService.findAll());
 
-const findById = createAsyncThunk('nft/findById', async (itemId: string) =>
-  itemService.findById(itemId)
-);
+const findById = createAsyncThunk('nft/findById', async (itemId: string, { dispatch }) => {
+  const item = await itemService.findById(itemId);
```
Comment (mmiranda-stackit) [inline src/features/leda-nft/store/leda-nft.actions.ts:47]: Avoid await if you don't need it

File: src/components/item-details/place-bet.tsx [hunk 6]
```diff
@@ -26,7 +26,7 @@ const PlaceBet = ({ highestBid, actionDate, btnColor, className, item }: Props)
           Buy
         </Button>
       </div>
-      <PlaceBidModal show={showBidModal} handleModal={handleBidModal} item={item} />
+      <PlaceBidModal show={showBidModal} handleModal={handleBidModal} />
```
Comment (mmiranda-stackit) [inline src/components/item-details/place-bet.tsx:29]: Let's change the 'bid' name for something more meaningful (we are not bidding here)

**Review by mmiranda-stackit — CHANGES_REQUESTED** [review]

**Review by mmiranda-stackit — APPROVED** [review]

**Pull request merged**